File: examples/TemplateEditExample.rb
require "json"
require "dropbox-sign"

Dropbox::Sign.configure do |config|
    config.username = "YOUR_API_KEY"
    # config.access_token = "YOUR_ACCESS_TOKEN"
end

template_edit_request = Dropbox::Sign::TemplateEditRequest.new
template_edit_request.cc_roles = [
    "Role 1",
    "Role 2",
]

begin
    response = Dropbox::Sign::TemplateApi.new.template_edit(
        "f57db65d3f933b5316d398057a36176831451a35", # template_id
        template_edit_request,
    )

    p response
rescue Dropbox::Sign::ApiError => e
    puts "Exception when calling TemplateApi#template_edit: #{e}"
end

File: examples/TemplateEditExample.java
package com.dropbox.sign_sandbox;

import com.dropbox.sign.ApiException;
import com.dropbox.sign.Configuration;
import com.dropbox.sign.api.*;
import com.dropbox.sign.auth.*;
import com.dropbox.sign.JSON;
import com.dropbox.sign.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class TemplateEditExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((HttpBasicAuth) config.getAuthentication("api_key")).setUsername("YOUR_API_KEY");
        // ((HttpBearerAuth) config.getAuthentication("oauth2")).setBearerToken("YOUR_ACCESS_TOKEN");

        var templateEditRequest = new TemplateEditRequest();
        templateEditRequest.ccRoles(List.of (
            "Role 1",
            "Role 2"
        ));

        try
        {
            var response = new TemplateApi(config).templateEdit(
                "f57db65d3f933b5316d398057a36176831451a35", // templateId
                templateEditRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling TemplateApi#templateEdit");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}
